pkgs/native_assets_builder/lib/src/build_runner/build_runner.dart

```diff
@@ -538,7 +538,7 @@ ${e.message}
             [
               ...result.dependencies,
               // Also depend on the hook source code.
-              hookHashesFile.uri,
+              hookHashes.file.uri,
```

Comment:
Let's assume we did one build. Now I modify `hook/build.dart` by adding a few comments.

Next time we run a build, the system sees that hook sources changed and it will re-compile the hook, giving us a new dill file.

Now, since we only changed comments in the `hook/build.dart`, we will get an identical `.dill` file.

So the next step is running the hook. Though we can skip running the hook if none of the hook-reported deps changed and the `hook.dill` didn't change (which it didn't)
=> In this scenario we could avoid running the hook.

Though this code looks like we're going to re-run the hook, because we add the `hooks.d` file here (where hashes of sources changed, we we added some comments).

(One could argue this redundant run is ok because it's very uncommon, but from conceptual point we should only re-run things when needed)